simulators/main/event.py

```diff
@@ -0,0 +1,318 @@
+import json, logging
+from datetime import datetime
+from random import randint, uniform, choices
+from cumulocityAPI import CumulocityAPI
+from task import Task, PeriodicTask
+import interface
+
+cumulocityAPI = CumulocityAPI()
+log = logging.getLogger("events generation")
+
+
+class Event(interface.MachineType):
+    def __init__(self, model, shiftplans) -> None:
+        self.model = model
+        self.shiftplans = shiftplans
+        self.device_id = self.model.get("device_id")
+        self.locationId = self.model.get("locationId", "")
+        self.machine_up = False
+        self.shutdown = False
+        self.production_time_s = 0.0
+        self.last_production_time_update = datetime.timestamp(datetime.utcnow())
+        self.out_of_production_time_logged = False
+        self.event_definitions = self.model.get('events', [])
+        self.enabled = self.model.get('enabled', True)
+        if self.enabled:
+            self.tasks = list(map(self.__create_task, self.event_definitions))
+            self.production_speed_s = Event.get_production_speed_s(self)
+            log.debug(f'events: {self.event_definitions}')
+
+    def __create_task(self, definition):
```

Comment:
I think for create task there should also be a implementation where it is inside of the MachineSimulator, since this is a method that should be executed for both implementations